frontend/js/src/explore/artist-similarity/SimilarArtistsGraph.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState } from "react";
+import {
+  ResponsiveNetwork,
+  NodeProps,
+  NodeTooltipProps,
+  NetworkSvgProps,
+} from "@nivo/network";
+import { animated, to } from "@react-spring/web";
+import type { NodeType, LinkType, GraphDataType } from "./Data";
+
+interface GraphProps {
+  data: GraphDataType;
+  onArtistChange: (artist_mbid: string) => void;
+  background: string;
+}
+
+type OmitHeightWidth<T> = Omit<T, "height" | "width">;
+
+const MAX_LINES = 2;
+const MAX_WORD_LENGTH = 10;
+
+function CustomNodeComponent({
+  node,
+  animated: animatedProps,
+  onClick,
+  onMouseEnter,
+  onMouseMove,
+  onMouseLeave,
+}: NodeProps<NodeType>) {
+  return (
+    <animated.g
+      transform={to(
+        [animatedProps.x, animatedProps.y, animatedProps.scale],
+        (x, y, scale) => {
+          return `translate(${x},${y}) scale(${scale})`;
+        }
+      )}
+      onClick={onClick ? (event) => onClick(node, event) : undefined}
+      onMouseEnter={
+        onMouseEnter ? (event) => onMouseEnter(node, event) : undefined
+      }
+      onMouseMove={
+        onMouseMove ? (event) => onMouseMove(node, event) : undefined
+      }
+      onMouseLeave={
+        onMouseLeave ? (event) => onMouseLeave(node, event) : undefined
+      }
+      width={to([animatedProps.size], (size) => size / 2)}
+      height={to([animatedProps.size], (size) => size / 2)}
+    >
+      <animated.circle
+        data-testid={`node.${node.id}`}
+        r={to([animatedProps.size], (size) => size / 2)}
+        fill={animatedProps.color}
+        strokeWidth={animatedProps.borderWidth}
+        stroke={animatedProps.borderColor}
+        opacity={animatedProps.opacity}
+      />
+      <animated.text
+        textAnchor="middle"
+        dominantBaseline="alphabetic"
```

Comment:
FYI, this centers the text vertically as one would expect (but I suggested an alternative option above)
```suggestion
        dominantBaseline="central"
```